src/react/utils.js

```diff
@@ -575,8 +559,37 @@ export function evaluateComponentTreeBranch(
   }
 }
 
+// This function is mainly use to delete internal properties
+// on objects that we know are safe to access internally
+// such as ReactElements. Deleting here does not
+// emit change to modified bindings and is intended
+// for only internal usage – not for user-land code
+export function deleteProperty(object: ObjectValue | AbstractObjectValue, property: string | SymbolValue): void {
+  if (object instanceof AbstractObjectValue) {
+    let elements = object.values.getElements();
+    if (elements && elements.size > 0) {
+      object = Array.from(elements)[0];
+    }
+    invariant(object instanceof ObjectValue);
+  }
+  let binding;
+  if (typeof property === "string") {
+    binding = object.properties.get(property);
+  } else {
+    binding = object.symbols.get(property);
+  }
+  if (!binding) {
+    return;
+  }
+  binding.descriptor = undefined;
```

Comment:
Why do we delete the descriptor rather than the binding itself from the property map?

Reply:
This is the expected behaviour as this is how $Delete also works.